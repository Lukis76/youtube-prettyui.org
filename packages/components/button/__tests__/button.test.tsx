import * as React from "react";
import Button from "../src/button";
import { cleanup, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { test, expect, vi, describe, afterEach } from "vitest";

// Renders a button element
describe("Button", () => {
  afterEach(cleanup);
  test("should render correctly", () => {
    const wrapper = render(<Button>Click me</Button>);
    expect(() => wrapper.unmount()).not.toThrow();
  });
  test("ref should be forwarded", () => {
    const ref = React.createRef<HTMLButtonElement>();
    render(<Button ref={ref}>Click me</Button>);
    expect(ref.current).not.toBeNull();
    expect(ref.current).toBeInstanceOf(HTMLButtonElement);
    expect(ref.current).toHaveTextContent("Click me");
  });

  test("should render a button element with the provided props", () => {
    // Arrange
    const props = {
      className: "custom-button",
      onClick: vi.fn(),
    };

    // Act
    render(<Button {...props}>Click me</Button>);

    // Assert
    const buttonElement = screen.getByRole("button");
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toMatchSnapshot();
    expect(buttonElement).toHaveClass("custom-button");
    expect(buttonElement).toHaveTextContent("Click me");
    // expect(buttonElement).toHaveAttribute("onClick", props.onClick);
  });
});
