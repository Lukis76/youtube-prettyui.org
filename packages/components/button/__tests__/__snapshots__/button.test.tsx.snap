// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should render a button element with the provided props 1`] = `
<button
  class="custom-button"
>
  Click me
</button>
`;
